hcl2template/parser.go

```diff
@@ -137,6 +137,7 @@ func (p *Parser) Parse(filename string, varFiles []string, argVars map[string]st
 		Basedir:                 basedir,
 		Cwd:                     wd,
 		CorePackerVersionString: p.CorePackerVersionString,
+		HCPVars:                 map[string]cty.Value{},
```

Comment:
HCPVars is essentially how we set variables from the HCPHandler back into the configuration, for example now the iteration ID is propagated to the eval context, so it can be used in the configuration later on.